plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/RetryingJdbcClient.java

```diff
@@ -0,0 +1,527 @@
+    public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<JdbcSortItem> sortOrder)
+    {
+        // there should be no remote database interaction
+        return delegate.supportsTopN(session, handle, sortOrder);
+    }
+
+    @Override
+    public boolean isTopNGuaranteed(ConnectorSession session)
+    {
+        // there should be no remote database interaction
+        return delegate.isTopNGuaranteed(session);
+    }
+
+    @Override
+    public boolean supportsLimit()
+    {
+        // there should be no remote database interaction
+        return delegate.supportsLimit();
+    }
+
+    @Override
+    public boolean isLimitGuaranteed(ConnectorSession session)
+    {
+        // there should be no remote database interaction
+        return delegate.isLimitGuaranteed(session);
+    }
+
+    @Override
+    public Optional<String> getTableComment(ResultSet resultSet)
+            throws SQLException
+    {
+        // no retrying as it could be not idempotent operation
+        return delegate.getTableComment(resultSet);
+    }
+
+    @Override
+    public void setTableComment(ConnectorSession session, JdbcTableHandle handle, Optional<String> comment)
+    {
+        // no retrying as it could be not idempotent operation
+        delegate.setTableComment(session, handle, comment);
+    }
+
+    @Override
+    public void setColumnComment(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle column, Optional<String> comment)
+    {
+        // no retrying as it could be not idempotent operation
+        delegate.setColumnComment(session, handle, column, comment);
+    }
+
+    @Override
+    public void addColumn(ConnectorSession session, JdbcTableHandle handle, ColumnMetadata column)
+    {
+        // no retrying as it could be not idempotent operation
+        delegate.addColumn(session, handle, column);
+    }
+
+    @Override
+    public void dropColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle column)
+    {
+        // no retrying as it could be not idempotent operation
+        delegate.dropColumn(session, handle, column);
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newColumnName)
+    {
+        // no retrying as it could be not idempotent operation
+        delegate.renameColumn(session, handle, jdbcColumn, newColumnName);
+    }
+
+    @Override
+    public void setColumnType(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle column, Type type)
+    {
+        // no retrying as it could be not idempotent operation
+        delegate.setColumnType(session, handle, column, type);
+    }
+
+    @Override
+    public void dropNotNullConstraint(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle column)
+    {
+        // no retrying as it could be not idempotent operation
+        delegate.dropNotNullConstraint(session, handle, column);
```

Comment:
safe to retry but the code needs to handle exceptions if the constraint was dropped already (e.g. timeouts)

Reply:
So let leave it as follow. I will update the comment then.